vmdb/spec/models/miq_alert_spec.rb

```diff
@@ -309,5 +309,73 @@
     end
   end
 
+  context ".evaluate_hourly_timer" do
+    before do
+      MiqAlert.any_instance.stub(:validate => true)
+      @guid = MiqUUID.new_guid
+      MiqServer.stub(:my_guid).and_return(@guid)
+
+      @zone            = FactoryGirl.create(:zone)
+      @miq_server      = FactoryGirl.create(:miq_server, :guid => @guid, :zone => @zone)
+      MiqServer.stub(:my_server).and_return(@miq_server)
+
+      @ems = FactoryGirl.create(:ems_vmware, :zone => @zone)
+      @ems_other = FactoryGirl.create(:ems_vmware, :zone => FactoryGirl.create(:zone, :name => 'other'))
+      @alert      = FactoryGirl.create(:miq_alert, :enabled => true, :responds_to_events => "_hourly_timer_")
+      @alert_prof = FactoryGirl.create(:miq_alert_set, :description => "Alert Profile for Alert Id: #{@alert.id}")
+      @alert_prof.add_member(@alert)
+    end
+
+    it "evaluates for ext_management_system" do
+      @alert.db = "ExtManagementSystem"
+      @alert.save!
```

Comment:
You can do this in one line with `@alert.update_attributes(:db => "ExtManagementSystem")`